### PR TITLE
Add draft of README

### LINKED ISSUE #21: Add README
Add description of project
Add instructions on how to properly use the CI-server
Add assessment and documentation of the team
Add list of contributions

### PR DESCRIPTION
**Pull request by filippanilsson**
Fix #21 

Add description of project, instructions on how to use the CI server, assessment of team, and two contributions

### CODE REVIEW

**Review by Frequinzy — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -1,2 +1,103 @@
...
+## Overview
+The CI server is a Maven project and contains three top folders inside the DD2480-CI project folder:
+* **src**: contains the source code of the server.
+  * **main**:
+    * **BuildStatus**: An enum containing the different types of commit statuses.
+    * **ContinuousIntegrationServer**: Houses the server, based on the skeleton found [here](https://github.com/KTH-DD2480/smallest-java-ci).
+    * **GitRepo**: Clones the repo to the server
+    * **GitStatusUpdate**: Sends a POST request to the GitHub REST API to update a commit’s status
+    * **MavenInvokerBuilder**: Executes the build for Maven repository and retrieves the build output and result status
+    * **RequestParser**: Checks if an HTTP request represents a GitHub push event and extracts information needed for build and notification.
```
Comment (annamaartensson) [inline README.md:18]: Missing BuildHistoryManager and TestAutomationHandler

File: README.md [hunk 2]
```diff
@@ -1,2 +1,106 @@
...
+## Running the server ##
+1. Set up port forwarding and Webhook for your repo
+2. run `mvn test` in the root of your project to trigger the building process locally
+3. run `mvn clean compile exec:java` to start the server
+4. Go to `localhost:8012` in your browser
+5. Push a commit to your repo
+
+## Features
+
+### Compilation
```
Comment (Tian0602) [inline README.md:50]: Our server automatically tests all corresponding repo tests using Maven Invoker Builder. The compilation class generates build results and logs.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,106 @@
...
+
+## Features
+
+### Compilation 
+...
+
+### Cloning the repo
+Our server will automatically clone and checkout the correct commit in the repo using the method cloneRepo and checkoutCommit, implemented with Jgit in the GitRepo.java file. The cloned repo will be stored in a temporary folder which will be deleted once the CI server is finished.
+
+### Execution of automated tests
```
Comment (Tian0602) [inline README.md:56]: Our server will analyze the HTTP request received from the Webhook, extract the branch, clone URL, and commit IDs. These details will be used to clone the git repository and add build history. The compilation process will then test the repository and update the relevant notification and history log accordingly.

**Review by Tian0602 — APPROVED** [review]
LGTM

**Pull request merged**